### PR TITLE
Manage static properties using a staticProvider extension point

### LINKED ISSUE #44: Manage static properties using a staticProvider extension point
As we continue to add static properties for MicroProfile specs it would be nice to improve the existing propertiesProvider extension point to load static property JSON files using something like this:

```xml
<staticProvider resource="/properties/metrics.json">
   <enabledWhen>
         <typeOnClasspath>org.eclipse.microprofile.metrics.MetricID</typeOnClasspath>
         ...
   </enabledWhen>
</staticProvider>
```

See: https://github.com/redhat-developer/quarkus-ls/issues/247

### PR DESCRIPTION
**Pull request by JessicaJHee**
Fixes #44 

Signed-off-by: Jessica He <jhe@redhat.com>

### CODE REVIEW

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticProviderRegistry.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        return INSTANCE;
+    }
+
+    private StaticProviderRegistry() {
+        super();
+    }
+
+    @Override
+    public String getProviderExtensionId() {
+        return "staticProviders";
```
Comment (datho7561) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticProviderRegistry.java:37]: I think it makes more sense to call this extension point `staticPropertyProviders`, just to be extra clear.

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProvider.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.eclipse.lsp4mp.jdt.core.SearchContext;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTTypeUtils;
+
+/**
+ * MicroProfile static properties provider.
+ *
+ */
+public class StaticPropertyProvider extends AbstractStaticPropertiesProvider {
+
+    String type;
```
Comment (rgrunber) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProvider.java:29]: Can this be made private ? Looks like you only is it for `isAdaptedFor(..)`. If you ever needed to use it outside this class, just create a getter and keep it private.

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProviderRegistry.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import org.eclipse.lsp4mp.jdt.core.MicroProfileCorePlugin;
+
+/**
+ * Registry to hold static providers for an extension point
+ *
+ */
+public class StaticPropertyProviderRegistry implements IRegistryChangeListener {
+
+    public StaticPropertyProviderRegistry() {
+        super();
```
Comment (rgrunber) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProviderRegistry.java:36]: I don't think you need `super()` here. It's just implementing an interface. There's no parent class that with a no-arg constructor to be called, unless I'm missing something.

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProvider.java [hunk 4]
```diff
@@ -1,39 +1,56 @@
...
-        super(MicroProfileCorePlugin.PLUGIN_ID, "/static-properties/mp-config-metadata.json");
-    }
-
-    @Override
-    protected boolean isAdaptedFor(SearchContext context, IProgressMonitor monitor) {
-        // MicroProfile config properties are always available with MicroProfile
-        return true;
-    }
-}
+/*******************************************************************************
```
Comment (datho7561) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProvider.java:1]: Replace spaces with tabs (sorry this is an IDE setting I accidentally modified during one of my previous PRs)

File: microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProviderRegistry.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import org.eclipse.core.runtime.IRegistryChangeEvent;
+import org.eclipse.core.runtime.IRegistryChangeListener;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.lsp4mp.jdt.core.MicroProfileCorePlugin;
+
+/**
+ * Registry to hold static providers for an extension point
+ *
+ */
+public class StaticPropertyProviderRegistry implements IRegistryChangeListener {
```
Comment (datho7561) [inline microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/StaticPropertyProviderRegistry.java:33]: Replace spaces with tabs

**Review by datho7561 — CHANGES_REQUESTED** [review]
A few comments on some code formatting, but it appears to be working well! Thanks, Jessica!

**Review by datho7561 — APPROVED** [review]
Looks good, thanks Jessica!

**Pull request merged**